src/coreComponents/physicsSolvers/wavePropagation/WaveSolverUtils.hpp

```diff
@@ -0,0 +1,227 @@
+    real32 const a2 = 1.0 - a1;
+
+    if( nsamplesSeismoTrace > 0 )
+    {
+      forAll< parallelDevicePolicy< 32 > >( receiverConstants.size( 0 ), [=] GEOSX_HOST_DEVICE ( localIndex const ircv )
+      {
+        if( receiverIsLocal[ircv] == 1 )
+        {
+          varAtReceivers[iSeismo][ircv] = 0.0;
+          real32 vtmp_np1 = 0.0;
+          real32 vtmp_n = 0.0;
+          for( localIndex inode = 0; inode < receiverConstants.size( 1 ); ++inode )
+          {
+            vtmp_np1 += var_np1[receiverNodeIds[ircv][inode]] * receiverConstants[ircv][inode];
+            vtmp_n += var_n[receiverNodeIds[ircv][inode]] * receiverConstants[ircv][inode];
+          }
+          // linear interpolation between the pressure value at time_n and time_(n+1)
+          varAtReceivers[iSeismo][ircv] = a1*vtmp_n + a2*vtmp_np1;
+        }
+      } );
+    }
+
+    // TODO DEBUG: the following output is only temporary until our wave propagation kernels are finalized.
+    // Output will then only be done via the previous code.
+    if( iSeismo == nsamplesSeismoTrace - 1 )
+    {
+      forAll< serialPolicy >( receiverConstants.size( 0 ), [=] ( localIndex const ircv )
+      {
+        if( outputSeismoTrace == 1 )
+        {
+          if( receiverIsLocal[ircv] == 1 )
+          {
+            // Note: this "manual" output to file is temporary
+            //       It should be removed as soon as we can use TimeHistory to output data not registered on the mesh
+            // TODO: remove saveSeismo and replace with TimeHistory
+            std::ofstream f( GEOSX_FMT( "seismoTraceReceiver{:03}.txt", ircv ), std::ios::app );
+            for( localIndex iSample = 0; iSample < nsamplesSeismoTrace; ++iSample )
+            {
+              f << iSample << " " << varAtReceivers[iSample][ircv] << std::endl;
+            }
+            f.close();
+          }
+        }
+      } );
+    }
+  }
+
+  /**
+   * @brief Check if the source point is inside an element or not
+   */
+
+  /**
+   * @brief Check if the source point is inside an element or not
+   * @param numFacesPerElem number of face on an element
+   * @param elemCenter array containing the center of the elements
+   * @param faceNormal array containing the normal of all faces
+   * @param faceCenter array containing the center of all faces
+   * @param elemsToFaces map to get the global faces from element index and local face index
+   * @param coords coordinate of the point
+   * @return true if coords is inside the element
+   */
+
+  GEOSX_HOST_DEVICE
+  static bool
+  locateSourceElement( real64 const numFacesPerElem,
+                       real64 const (&elemCenter)[3],
+                       arrayView2d< real64 const > const faceNormal,
+                       arrayView2d< real64 const > const faceCenter,
+                       arraySlice1d< localIndex const > const elemsToFaces,
+                       real64 const (&coords)[3] )
```

Comment:
In future refactorings, if you want, you can put this function in `mesh/utilities/ComputationalGeometry.hpp` and merge it with `isPointInsidePolyhedron`.